test.js

```diff
@@ -26,3 +26,11 @@ test('sync with custom shell', t => {
 	t.true('HOME' in env);
 	t.false('' in env);
 });
+
+test('sync with custom shell throws on non-executable', async t => {
+	t.throws(() => m.sync('non-executable'));
+});
+
+test('async with custom shell throws on non-executable', async t => {
+	await t.throws(m('non-executable'));
```

Comment:
Wow, ✨ magic ✨

I guess that it's advisable to `await` it if you have stuff happening after though? Since otherwise the code path will continue even if there was an error at that point...